### DISCUSSION TITLE
v-otp-input add rules and error-messages for handling validation error messages

### DISCUSSION THREAD

**Discussion by adnanaz** [vuetifyjs/vuetify#14753] [Ideas]
I saw at v-otp-input APIdidnt have rules and error-messages when i need to put validation on it, and then i have to make customlly, i think it would be same with v-text-input with error-handling show for consistency thankyou.

**Comment by ggedde** [comment]
I agree,
When I went to use v-otp-input yesterda, I was flustered as I could not get rules to work. Then I reviewed the documentation and GitHub issues to find out that it was not implemented and not planned to be :(

I think any input component should allow for rules. This would make things more consistent and dependable.

PS. the custom rules validation in Vuetify is one of my favorite features. Thanks Vuetify team.
I really wish I can use it more often and for any form element.

I do think that maybe this was discontinued as some might think each input field should or should not get its own rules or message fields. I don't think that should be the case. It should be for the entire group of inputs. Basically, the rules should validate the model value as a single value, just as it does for all other input components. If you need individual validation then I agree that should be custom built by the developer and not Vuetify.

Vuetify Team, any chance we can have this reconsidered?

**Comment by MtDalPizzol** [comment]
Just hit this... It makes no sense to me that the VOtpInput has an `error` prop just like other form components, but its missing the `error-messages`. What's the reasoning behind this?

**Comment by craigrileyuk** [comment]
Yeah, we have to wrap the Vuetify component in a custom component to add all of the validation feedback to the component.

**Comment by cojohnson1999** [comment]
This is also an issue my team has run into.  It feels strange that this specific input field isn't given `error-messages` support considering it is an input field that should seemingly have and support validations.

It would be great for the Vuetify team to reconsider this for consistency purposes.
